packages/raystack/table/FilteredChip.tsx

```diff
@@ -29,26 +30,43 @@ interface FilterValuesProps {
   columnType?: columnTypes;
   onValueChange?: (value: FilterValue) => void;
   options?: TableColumnMetadata[];
+  filterOperation?: FilterOperation;
 }
 
 const FilterValues = ({
   columnType = filterValueTypeMap.text,
   options = [],
   onValueChange,
+  filterOperation,
+  ...props
 }: FilterValuesProps) => {
-  const [value, setValue] = useState("");
-  const valueType: filterValueType =
-    columnType === columnTypesMap.select
-      ? filterValueTypeMap.select
-      : filterValueTypeMap.text;
+  const [value, setValue] = useState<FilterValue>({
+    date: new Date(),
+    dateRange: {
+      to: new Date(),
+      from: new Date(),
+    },
+  });
+
+  const valueType: filterValueType = filterOperation?.component
+    ? filterOperation?.component
+    : columnType === columnTypesMap.date
+    ? filterValueTypeMap.datePicker
+    : columnType === columnTypesMap.select
+    ? filterValueTypeMap.select
+    : filterValueTypeMap.text;
 
   useEffect(() => {
-    if (value && onValueChange) {
-      onValueChange({ value });
+    if (onValueChange) {
+      onValueChange(value);
     }
   }, [value]);
+
   return valueType === filterValueTypeMap.select ? (
```

Comment:
For readability of this entire section, can we use a switch case here, and get the different components we return via functions?
```
switch(valueType) {
  case filterValueTypeMap.select {
     return getSelectFilter(arg1, arg2, arg3, ....)
  }
  case filterValueTypeMap.datePicker {
    return getDatePicker(arg1.....)
  }
}
```

Actually, we could even use the factory pattern here, depending on the types of inputs we expect to receive.

Reply:
Using functions inside the React component return statement is like an anti pattern.
We can create component like `SelectFilter`, `DateRangeFilter` etc
https://unicorn-utterances.com/posts/functions-are-killing-react-performance

Reply:
Got it. Does it make sense to convert the inner logic to components instead of functions? [Like it is mentioned here in the eslint github repo](https://github.com/jsx-eslint/eslint-plugin-react/issues/578)?

So code becomes:
```
switch(valueType) {
  case filterValueTypeMap.select {
     <MySelectComponent prop1={} prop2={} />
  }
  case filterValueTypeMap.datePicker {
    <MyDatePickerComponent prop1={} prop2={} />
  }
}
```